### PR TITLE
Fix muclight hook call

### PR DESCRIPTION
**Pull request by bartekgorny**
Fixed invalid call to `filter_room_packet` hook - handlers for that hook have different arity.

The call never really makes it to handlers because it is made from muc_light, while the only handler for this hook is added by mam, but still it should be done right.

### CODE REVIEW

File: apps/ejabberd/src/mod_muc_light_codec_legacy.erl [hunk 1]
```diff
@@ -99,7 +107,7 @@ decode_message(#xmlel{ attrs = Attrs, children = Children }) ->
 -spec decode_message_by_type(Type :: {binary(), binary()} | false,
```
Comment (gadgetci) [inline apps/ejabberd/src/mod_muc_light_codec_legacy.erl]: According to **Elvis**:
> The spec in line 107 uses a record, please define a type for the record and use that instead.

**Comment by gadgetci** [comment]
**compiler** failed with exit status: 1

File: apps/ejabberd/include/mod_muc_light.hrl [hunk 2]
```diff
@@ -121,3 +121,6 @@
                                   | {set, #config{}, AffUsers :: aff_users()}
                                   | {error, bad_request}.
 
+-type msg() :: #msg{}.
+
+-type config() :: #config{}.
```
Comment (gadgetci) [inline apps/ejabberd/include/mod_muc_light.hrl]: According to **Compiler**:
> type config() already defined

**Comment by michalwski** [comment]
Is it possible to have a test case showing the problem? This issue happened only because we didn't have it tested.

**Pull request merged**